logical/logical.go

```diff
@@ -2,6 +2,32 @@ package logical
 
 import log "github.com/mgutz/logxi/v1"
 
+// BackendType is the type of backend that is being implemented
+type BackendType uint32
+
+// The these are the types of backends that can be derived from
+// logical.Backend
+const (
+	TypeUnknown     BackendType = 0 // This is also the zero-value for BackendType
+	TypeLogical     BackendType = 1
+	TypeCredential  BackendType = 2
+	TypePassthrough BackendType = 3
```

Comment:
I bumped across this when I was trying to implement a `generic` plugin backend. We can revert back to the old behavior, and revisit if this comes up again.